force-app/main/default/classes/FuzzyPicklists.cls

```diff
@@ -5,20 +5,19 @@ public class FuzzyPicklists {
         List<FuzzyPicklist__mdt> fps = getFuzzyPicklists(recordIds);
         //find all relevant FuzzyPicklistValues and FuzzyPicklistPossibilities
         List<FuzzyPicklistValue__mdt> fpvs = getFuzzyPicklistValuesWithPossibilities(fps);
-        Map<Id, sObject> recordsToUpdate = new Map<Id, sObject>();
         //find full info from all IDs with
         //dynamic soql assembling free text field and picklist field as defined in fp
         String query = buildBaseQueryString(fps,recordIds)+' WHERE Id IN :recordIds';
         List<sObject> records = Database.query(query);
+
+        Map<Id, sObject> recordsToUpdate = new Map<Id, sObject>();
         //for each FP, pass to matchPicklist
         for(FuzzyPicklist__mdt fp : fps){
             for(sObject rec : matchPicklist(fp, fpvs, records)){
                 recordsToUpdate.put(rec.Id,rec);
             }
         }
-        if(recordsToUpdate != null && recordsToUpdate.size()>0){
-            update recordsToUpdate.values();
-        }
+        update recordsToUpdate.values();
```

Comment:
No need to check if there any any record in the Map. Apex is smart enough to do this for you.